File: AmpedBiz/AmpedBiz.Web/src/common/models/role.ts
export type Role = {
  id: string;
  name: string;
  assigned: boolean;
}

export const role = {
  admin: <Role>{
    id: 'A',
    name: 'Admin'
  },
  manager: <Role>{
    id: 'M',
    name: 'Manager'
  },
  salesclerk: <Role>{
    id: 'S',
    name: 'Salesclerk'
  },
  cashier: <Role>{
    id: 'C',
    name: 'Cashier'
  },
  warehouseman: <Role>{
    id: 'W',
    name: 'Warehouseman'
  },
  unknown: <Role>{
    id: 'UR',
    name: 'Unknown'
  },  
  all: () => [
    role.admin,
    role.manager,
    role.salesclerk,
    role.warehouseman,
  ],
}

File: AmpedBiz/AmpedBiz.Core/Users/Role.cs
using System.Collections.Generic;

namespace AmpedBiz.Core.Users
{
    public class Role : Entity<string, Role>
    {
        public virtual string Name { get; set; }

        public Role() : base(default(string)) { }

        public Role(string id, string name = null) : base(id)
        {
            this.Name = name;
        }

        public static readonly Role Admin = new Role("A", "Admin");

        public static readonly Role Manger = new Role("M", "Manager");

        public static readonly Role Salesclerk = new Role("S", "Salesclerk");

        public static readonly Role Warehouseman = new Role("W", "Warehouseman");

        public static IEnumerable<Role> All = new [] 
        {
            Role.Admin,
            Role.Manger,
            Role.Salesclerk,
            Role.Warehouseman
        };
    }
}
